Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -0,0 +1,482 @@
+
+        /// <summary>
+        /// This is the radius of either the Cylinder or Sphere mapping and is ignored when using the plane mapping.
+        /// </summary>
+        [Range(0.05f, 5.0f)]
+        [Tooltip("Radius for the sphere or cylinder")]
+        [SerializeField]
+        private float radius = 2f;
+
+        /// <summary>
+        /// Number of rows per column, column number is automatically determined
+        /// </summary>
+        [Tooltip("Number of rows per column")]
+        [SerializeField]
+        private int rows = 3;
+
+        /// <summary>
+        /// Width of the cell per object in the collection.
+        /// </summary>
+        [Tooltip("Width of cell per object")]
+        [SerializeField]
+        private float cellWidth = 0.5f;
+
+        /// <summary>
+        /// Height of the cell per object in the collection.
+        /// </summary>
+        [Tooltip("Height of cell per object")]
+        [SerializeField]
+        private float cellHeight = 0.5f;
+
+        private Mesh sphereMesh;
+        private Mesh cylinderMesh;
+        private int columns;
+        private float width;
+        private float height;
+        private float circumferenc;
+        private Vector2 halfcell;
+        #endregion
+
+        #region public accessors
+        public ObjectOrientationSurfaceType SurfaceType
+        {
+            get { return surfaceType; }
+            set { surfaceType = value; }
+
+        }
+        public OrientationType OrientType
+        {
+            get { return orientType; }
+            set { orientType = value; }
+
+        }
+        public LayoutType Layout
+        {
+            get { return layout; }
+            set { layout = value; }
+
+        }
+        public float Radius
+        {
+            get { return radius; }
+            set { radius = value; }
+
+        }
+        public int Rows
+        {
+            get { return rows; }
+            set { rows = value; }
+
+        }
+        public float CellWidth
+        {
+            get { return cellWidth; }
+            set { cellWidth = value; }
+
+        }
+        public float CellHeight
+        {
+            get { return cellHeight; }
+            set { cellHeight = value; }
+
+        }
+
+        /// <summary>
+        /// Reference mesh to use for rendering the sphere layout
+        /// </summary>
+        public Mesh SphereMesh
+        {
+            get { return sphereMesh; }
+            set { sphereMesh = value; }
+
+        }
+
+        /// <summary>
+        /// Reference mesh to use for rendering the cylinder layout
+        /// </summary>
+        public Mesh CylinderMesh
+        {
+            get { return cylinderMesh; }
+            set { cylinderMesh = value; }
+        }
+
+        public float Width
+        {
+            get { return width; }
+        }
+
+        public float Height
+        {
+            get { return height; }
+        }
+        #endregion
```

Comment:
we've been adding the name of the region at the beginning and end to help identify them better.